pkg/services/ngalert/notifier/silence_svc.go

```diff
@@ -125,3 +143,86 @@ func (s *SilenceService) DeleteSilence(ctx context.Context, user identity.Reques
 
 	return nil
 }
+
+// WithAccessControlMetadata adds access control metadata to the given SilenceWithMetadata.
+func (s *SilenceService) WithAccessControlMetadata(ctx context.Context, user identity.Requester, silencesWithMetadata ...*models.SilenceWithMetadata) error {
+	silences := make([]*models.Silence, 0, len(silencesWithMetadata))
+	for _, silence := range silencesWithMetadata {
+		silences = append(silences, silence.Silence)
+	}
+	permissions, err := s.authz.SilenceAccess(ctx, user, silences)
+	if err != nil {
+		return err
+	}
+
+	if len(permissions) != len(silences) {
+		s.log.Error("failed to get metadata for all silences")
```

Comment:
perhaps Warn would be enough. WDYT? Error are suitable in a case when the normal program flow was interrupted.